[c10d] Add a logger for all nccl collectives with its time duration when completed

fduwjj opened the pull request:
Summary: We want to build a logging table for tracking the collective time spent on GPU for all internal workloads. Since we have a cudaEventQuery for both the start and end of a collective (We rolled out ECudaEventStart (enableTiming) fully already), we plan to add this logging table inside the watchdog of PyTorch ProcessGroupNCCL so that we get to know the duration of collectives.

Test Plan:
CI + dry run.

Rollback Plan:

Differential Revision: D76552340


cc @H-Huang @awgu @wanchaol @fegin @wz337 @wconstab @d4l3k

fegin reviewed (approved):
PR is already accepted internally.

fegin reviewed (approved):
PR is already accepted internally.

Commit:
[c10d] Add a logger for all nccl collectives with its time duration when completed (#156008)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/156008

We want to build a logging table for tracking the collective time spent on GPU for all internal workloads. Since we have a cudaEventQuery for both the start and end of a collective (We rolled out ECudaEventStart (enableTiming) fully already), we plan to add this logging table inside the watchdog of PyTorch ProcessGroupNCCL so that we get to know the duration of collectives.

Test Plan:
CI + dry run.

Rollback Plan:

Reviewed By: fegin, H-Huang, uthakore

Differential Revision: D76552340
